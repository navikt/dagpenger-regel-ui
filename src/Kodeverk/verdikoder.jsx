export const inntektTyper = [
  'Aksjer/grunnfondsbevis til underkurs',
  'Annen arbeidsinntekt',
  'Arbeidsopphold kost',
  'Arbeidsopphold losji',
  'Beregnet skatt',
  'Besøksreiser hjemmet annet',
  'Besøksreiser hjemmet kilometergodtgjørelse bil',
  'Betalt utenlandsk skatt',
  'Bil',
  'Bolig',
  'Bonus',
  'Bonus fra forsvaret',
  'Elektronisk kommunikasjon',
  'Fast bilgodtgjørelse',
  'Faste tillegg',
  'Fastlønn',
  'Feriepenger',
  'Fond for idrettsutøvere',
  'Foreldrepenger fra folketrygden',
  'Helligdagstillegg',
  'Honorar, akkord, prosent eller provisjonslønn',
  'Hyretillegg',
  'Innbetaling til utenlandsk pensjonsordning',
  'Kilometergodtgjørelse bil',
  'Kommunal omsorgslønn og fosterhjemsgodtgjørelse',
  'Kost (dager)',
  'Kost (døgn)',
  'Kostbesparelse i hjemmet',
  'Losji',
  'Lønn mv som ikke er skattepliktig i Norge fra utenlandsk diplomatisk eller konsulær stasjon',
  'Lønn og godtgjørelse til dagmamma eller praktikant som passer barn i barnets hjem',
  'Lønn og godtgjørelse til privatpersoner for arbeidsoppdrag i oppdragsgivers hjem',
  'Lønn og godtgjørelse utbetalt av veldedig eller allmennyttig institusjon eller organisasjon',
  'Lønn til verge fra Fylkesmannen',
  'Opsjoner',
  'Overtidsgodtgjørelse',
  'Reise annet',
  'Reise kost',
  'Reise losji',
  'Rentefordel lån',
  'Skattepliktig del av visse typer forsikringer',
  'Sluttvederlag',
  'Smusstillegg',
  'Stipend',
  'Styrehonorar og godtgjørelse i forbindelse med verv',
  'Svangerskapspenger',
  'Timelønn',
  'Trekk i lønn for ferie',
  'Uregelmessige tillegg knyttet til arbeidet tid',
  'Uregelmessige tillegg knyttet til ikke-arbeidet tid',
  'Yrkebil tjenestligbehov kilometer',
  'Yrkebil tjenestligbehov listepris',
  'Dagpenger ved arbeidsløshet',
  'Dagpenger til fisker',
  'Dagpenger til fisker som bare har hyre',
  'Sykepenger til fisker',
  'Sykepenger til fisker som bare har hyre',
  'Annen næringsinntekt',
  'Hyre - Annet',
  'Hyre - Bonus',
  'Hyre - Faste tillegg',
  'Hyre - Fastlønn',
  'Hyre - Feriepenger',
  'Hyre - Helligdagstillegg',
  'Hyre - Overtidsgodtgjørelse',
  'Hyre - Sluttvederlag',
  'Hyre - Timelønn',
  'Hyre - Uregelmessige tillegg knyttet til arbeidet tid',
  'Hyre - Uregelmessige tillegg knyttet til ikke-arbeidet tid',
  'Lott det skal beregnes trygdeavgift av',
  'Trekk i lønn for ferie - Hyre',
  'Vederlag lott',
  'Sykepenger fra folketrygden',
  'Sykepenger fra næring',
  'Sykepenger til dagmamma',
  'Sykepenger til jord- og skogbrukere',
  'Tiltak - Annet',
  'Tiltak - Bonus',
  'Tiltak - Faste tillegg',
  'Tiltak - Fastlønn',
  'Tiltak - Feriepenger',
  'Tiltak - Helligdagstillegg',
  'Tiltak - Overtidsgodtgjørelse',
  'Tiltak - Sluttvederlag',
  'Tiltak - Timelønn',
  'Tiltak - Uregelmessige tillegg knyttet til arbeidet tid',
  'Tiltak - Uregelmessige tillegg knyttet til ikke-arbeidet tid',
  'Trekk i lønn for ferie - Tiltak',
];

export const aktoerType = [
  'ORGANISASJON',
  'PRIVATPERSON',
];

export default { inntektTyper, aktoerType };
