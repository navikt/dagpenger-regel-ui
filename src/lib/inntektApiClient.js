import axios from 'axios';

export const getInntekt = async (request) => await axios({
  method: 'post',
  url: `/api/v1/inntekt/uklassifisert`,
  data: request,
}).then(response => response)
  .catch(error => error);

export const lagreInntekt = async (request) => await axios({
  method: 'post',
  url: `/api/v1/inntekt/uklassifisert/update`,
  data: request,
}).then(response => response)
  .catch(error => error);
