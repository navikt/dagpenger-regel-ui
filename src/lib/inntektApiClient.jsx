import axios from 'axios';

const apiUri = '/api/v1/inntekt';
const apiUklassifisertUri = `${apiUri}/uklassifisert`;
const apiUncachedUri = `${apiUklassifisertUri}/uncached`;

export const getInntekt = async (uri) => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUklassifisertUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`,
    });
  } catch (error) {
    return error;
  }
};


export const getUncachedInntekt = async (uri) => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUncachedUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`,
    });
  } catch (error) {
    return error;
  }
};

// todo post til uncached hvis henter nye opplysninger
// todo legge til ${request.aktørId}/${request.vedtakId}/${request.beregningsDato}`,
export const lagreInntekt = async (request, isUncached, uri) => {
  try {
    return await axios({
      method: 'post',
      url: isUncached
        ? `${apiUncachedUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`
        : `${apiUklassifisertUri}/${uri.aktørId}/${uri.vedtakId}/${uri.beregningsDato}`,
      data: request,
    });
  } catch (error) {
    return error;
  }
};

export const getVerdikoder = async () => {
  try {
    return await axios({
      method: 'get',
      url: `${apiUri}/verdikoder`,
    });
  } catch (error) {
    return error;
  }
};

