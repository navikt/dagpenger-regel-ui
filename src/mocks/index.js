async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen({ onUnhandledRequest: "warn" });
  } else {
    const { worker } = await import("./browser");
    worker.start({ onUnhandledRequest: "warn" });
  }
}

initMocks();

export {};
