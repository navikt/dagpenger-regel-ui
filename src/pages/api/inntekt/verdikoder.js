import { verdikoder } from "../../../lib/api/inntekt";

export default async function handler(req, res) {
  const data = await fetch(verdikoder());

  res.json(await data.json());
}
