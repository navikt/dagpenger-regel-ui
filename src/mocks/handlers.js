import { rest } from "msw";
import verdikoder from "./fixtures/verdikoder.json";
import inntektUklassifisert from "./fixtures/inntekt-uklassifisert.json";
import enhetsregisteret from "./fixtures/enhetsregisteret.json";

export const handlers = [
  rest.get("/api/inntekt/verdikoder", (req, res, ctx) =>
    res(ctx.json(verdikoder))
  ),
  rest.get("/api/inntekt", (req, res, ctx) =>
    res(ctx.json(inntektUklassifisert))
  ),
  rest.post("/api/inntekt", (req, res, ctx) =>
    res(ctx.json({ success: true }))
  ),
  rest.get("/api/enhet/:enhetId", (req, res, ctx) =>
    res(ctx.json(enhetsregisteret))
  ),
];
