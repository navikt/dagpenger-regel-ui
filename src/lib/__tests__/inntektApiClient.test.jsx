import { getInntekt, getVerdikoder, lagreInntekt } from "../inntektApiClient";
import verdikoder from "../../mocks/fixtures/verdikoder.json";
import { rest } from "msw";
import { server } from "../../mocks/server";

const baseURL = "http://localhost";

const apiUri = "/api/v1/inntekt";
const apiUklassifisertUri = `${apiUri}/uklassifisert`;
const apiUncachedUri = `${apiUklassifisertUri}/uncached`;

const params = {
  aktÃ¸rId: "111",
  kontekst: "vedtak",
  vedtakId: 12345,
  beregningsDato: "2019-05-01",
};

const isUncached = true;

describe("InntektApiClient", () => {
  it("Should handle 200 response", async () => {
    const expectedReturn = "data";

    const data = await getInntekt(params);
    expect(data.inntekt.arbeidsInntektMaaned).toHaveLength(6);
    expect(data).toMatchSnapshot();
  });

  it("Should handle 500 response", async () => {
    server.use(
      rest.get("/api/inntekt", (req, res, ctx) => res(ctx.status(500)))
    );

    await expect(getInntekt(params)).rejects.toThrow("Internal Server Error");
  });

  it("Should save inntekt with 200", async () => {
    const data = { field: "123" };
    const result = await lagreInntekt(data, isUncached, params);
    expect(result).toEqual({ success: true });
  });

  it("Should fail to save inntekt", async () => {
    const data = { field: "123" };
    server.use(
      rest.post("/api/inntekt/uncached", (req, res, ctx) =>
        res(ctx.status(500))
      )
    );

    await expect(lagreInntekt(data, isUncached, params)).rejects.toThrow(
      "Internal Server Error"
    );
  });

  it("Skal hente verdikoder", async () => {
    const result = await getVerdikoder();
    expect(result).toEqual(verdikoder);
  });
});
