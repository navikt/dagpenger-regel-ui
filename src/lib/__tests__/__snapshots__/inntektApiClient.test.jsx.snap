// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InntektApiClient Should handle 200 response 1`] = `
Object {
  "inntekt": Object {
    "arbeidsInntektMaaned": Array [
      Object {
        "aarMaaned": "2017-08",
        "arbeidsInntektInformasjon": Object {
          "inntektListe": Array [
            Object {
              "beloep": "0.00",
              "beskrivelse": "Total lønnsinntekt",
              "fordel": "kontantytelse",
              "header": "Total lønnsinntekt",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
              "utbetaltIMaaned": "2018-03",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
            },
          ],
        },
      },
      Object {
        "aarMaaned": "2017-07",
        "arbeidsInntektInformasjon": Object {
          "inntektListe": Array [
            Object {
              "beloep": "6410.26",
              "beskrivelse": "Total lønnsinntekt",
              "fordel": "kontantytelse",
              "header": "Total lønnsinntekt",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
              "utbetaltIMaaned": "2017-09",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
            },
            Object {
              "beloep": "41666.67",
              "beskrivelse": "Fastlønn",
              "fordel": "kontantytelse",
              "header": "Kontaktytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-09",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "48076.93",
              "beskrivelse": "Trekk i lønn for ferie",
              "fordel": "kontantytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-09",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
          ],
        },
      },
      Object {
        "aarMaaned": "2017-09",
        "arbeidsInntektInformasjon": Object {
          "inntektListe": Array [
            Object {
              "beloep": "54194.06",
              "beskrivelse": "Total lønnsinntekt",
              "fordel": "kontantytelse",
              "header": "Total lønnsinntekt",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
            },
            Object {
              "beloep": "48076.26",
              "beskrivelse": "Fastlønn",
              "fordel": "kontantytelse",
              "header": "Kontaktytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "52.00",
              "beskrivelse": "Overtidsgodtgjørelse",
              "fordel": "kontantytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "6065.80",
              "beskrivelse": "Timelønn",
              "fordel": "kontantytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
          ],
        },
      },
      Object {
        "aarMaaned": "2017-10",
        "arbeidsInntektInformasjon": Object {
          "inntektListe": Array [
            Object {
              "beloep": "36886.20",
              "beskrivelse": "Total lønnsinntekt",
              "fordel": "kontantytelse",
              "header": "Total lønnsinntekt",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
              "tilleggsinformasjon": Object {
                "kategori": "Nettoloennsordning",
              },
              "utbetaltIMaaned": "2018-04",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
            },
            Object {
              "beloep": "312.00",
              "beskrivelse": "Overtidsgodtgjørelse",
              "fordel": "kontantytelse",
              "header": "Kontaktytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "36574.20",
              "beskrivelse": "Timelønn",
              "fordel": "kontantytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
          ],
        },
      },
      Object {
        "aarMaaned": "2017-11",
        "arbeidsInntektInformasjon": Object {
          "inntektListe": Array [
            Object {
              "beloep": "39941.20",
              "beskrivelse": "Total lønnsinntekt",
              "fordel": "kontantytelse",
              "header": "Total lønnsinntekt",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2018-03",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "182.00",
              "beskrivelse": "Overtidsgodtgjørelse",
              "fordel": "kontantytelse",
              "header": "Kontantytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2018-03",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "39756.20",
              "beskrivelse": "Timelønn",
              "fordel": "kontantytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2018-03",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
          ],
        },
      },
      Object {
        "aarMaaned": "2019-05",
        "arbeidsInntektInformasjon": Object {
          "inntektListe": Array [
            Object {
              "beloep": "48569.00",
              "beskrivelse": "Total lønnsinntekt",
              "fordel": "kontantytelse",
              "header": "Total lønnsinntekt",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
              "tilleggsinformasjon": Object {
                "kategori": "Nettoloennsordning",
              },
              "utbetaltIMaaned": "2018-04",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "222222222",
              },
            },
            Object {
              "beloep": "44665.00",
              "beskrivelse": "Fastlønn",
              "fordel": "kontantytelse",
              "header": "Kontaktytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "44665.00",
              "beskrivelse": "Fastlønn",
              "fordel": "kontantytelse",
              "header": "Kontaktytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "333333333",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "333333333",
              },
            },
            Object {
              "beloep": "44665.00",
              "beskrivelse": "Fastlønn",
              "fordel": "kontantytelse",
              "header": "Kontaktytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "444444444",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "444444444",
              },
            },
            Object {
              "beloep": "366.00",
              "beskrivelse": "Elektronisk kommunikasjon",
              "fordel": "kontantytelse",
              "header": "Naturalytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
            Object {
              "beloep": "538.00",
              "beskrivelse": "Skattepliktig del av visse typer forsikring",
              "fordel": "kontantytelse",
              "informasjonsstatus": "InngaarAlltid",
              "inngaarIGrunnlagForTrekk": true,
              "inntektType": "LOENNSINNTEKT",
              "inntektskilde": "A-ordningen",
              "inntektsmottaker": Object {
                "aktoerType": "AKTOER_ID",
                "identifikator": "99999999999",
              },
              "inntektsperiodetype": "Maaned",
              "inntektsstatus": "LoependeInnrapportert",
              "leveringstidspunkt": "2019-02",
              "opplysningspliktig": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
              "utbetaltIMaaned": "2017-08",
              "utloeserArbeidsgiveravgift": true,
              "verdikode": "Fastlønn",
              "virksomhet": Object {
                "aktoerType": "ORGANISASJON",
                "identifikator": "111111111",
              },
            },
          ],
        },
      },
    ],
    "fraDato": "2016-06",
    "ident": Object {
      "aktoerType": "AKTOER_ID",
      "identifikator": "99999999999",
    },
    "tilDato": "2019-05",
  },
  "inntektId": Object {
    "id": "01D8G6FS9QGRT3JKBTA5KEE64C",
  },
  "inntektsmottaker": Object {
    "navn": "Navn Navnesen",
    "pnr": "12345678910",
  },
  "manueltRedigert": true,
  "redigertAvSaksbehandler": true,
  "timestamp": "2019-06-24T11:03:56.907821",
}
`;
